R/derive_adlb_params.R

```diff
@@ -0,0 +1,408 @@
+#'
+#' @param by_vars Grouping variables
+#'
+#'   Only variables specified in `by_vars` will be populated
+#'   in the newly created records.
+#'
+#'   `r admiral:::roxygen_param_by_vars()`
+#'
+#' @permitted [var_list]
+#'
+#' @param order Sort order
+#'
+#'   The observations are ordered by the specified order while looking for
+#'   consecutive results fulfilling the criteria to confirm a DIABETIC status.
+#'
+#'   If an expression is named, e.g., `exprs(LBDT = convert_dtc_to_dt(LBDTC), LBSEQ)`,
+#'   a corresponding variable (`LBDT`) is added to the additional dataset and used
+#'   for sorting. This variable is not included in the output dataset.
+#'
+#'   `r admiral:::roxygen_order_na_handling()`
+#'
+#' @permitted [var_list]
+#'
+#' @param hba1c_code HbA1c parameter code
+#'
+#'   The observations where `PARAMCD` equals the specified value are considered
+#'   as the HbA1c. It is expected that HbA1c is measured in mmol/mol or %.
+#'
+#' @permitted character value
+#'
+#' @param fpg_code Fasting Plasma Glucose parameter code
+#'
+#'   The observations where `PARAMCD` equals the specified value are considered
+#'   as the Fasting Plasma Glucose. It is expected that Fasting Plasma Glucose
+#'   is measured in mmol/L or mg/dL.
+#'
+#' @permitted character value
+#'
+#' @param set_values_to Variables to be set
+#'
+#'   The specified variables are set to the specified values for the new
+#'   observations. For example `exprs(PARAMCD = "GLYCSTT")` defines the parameter code
+#'   for the new parameter.
+#'
+#' @permitted List of variable-value pairs
+#'
+#' @param get_unit_expr An expression providing the unit of the parameter
+#'
+#'   The result is used to check the units of the input parameters.
+#'
+#' @permitted An expression which is evaluable in the input dataset
+#'   and results in a character value
+#'
+#' @param prediabetic_thresholds,diabetic_thresholds
+#'   Criteria for prediabetic and diabetic statuses
+#'
+#'   The specified thresholds are used to identify HbA1c and Fasting Plasma Glucose
+#'   results that corresponds to a certain glycemic status.
+#'
+#'   *Diabetic:*
+#'   - HbA1c: >=6.5% (48 mmol/mol)
+#'   - Fasting Plasma Glucose: >=126 mg/dL (7.0 mmol/L)
+#'
+#'   *Prediabetic:*
+#'   - HbA1c: >=5.7% (39 mmol/mol) but less than the threshold for *Diabetic*
+#'   - Fasting Plasma Glucose: >=100 mg/dL (5.6 mmol/L)
+#'
+#'   *Normoglycemic:*
+#'   - Any values not fulfilling the criteria above
+#'
+#'   Normally the criteria don’t need to be changed, but in case they do,
+#'   it can be done as follows:
+#'
+#'   ```
+#'   prediabetic_thresholds = list(
+#'     HBA1CHGB = xx.x,
+#'     GLUC = xx.x
+#'   ),
+#'   diabetic_thresholds = list(
+#'     HBA1CHGB = xx.x,
+#'     GLUC = xx.x
+#'   )
+#'   ```
+#'
+#'   where `xx.x` is a numeric threshold and names of the elements must match
+#'   `hba1c_code` and `fpg_code`.
+#'
+#' @permitted List of parameter-value pairs
+#'
+#' @details
+#' Implementation is based on ADA 2024.
```

Comment:
Should we try directly referencing the article? I.e. https://diabetesjournals.org/care/article/47/Supplement_1/S20/153954/2-Diagnosis-and-Classification-of-Diabetes. I am unsure how this is done in R packages, but can look it up.